src/App.tsx

```diff
@@ -15,81 +14,136 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-enum SortType {
-  NONE,
-  ALPHABET,
-  LENGTH,
-}
+class App extends React.Component<{}, State> {
+  state = {
+    goods: goodsFromServer,
+    start: false,
+    isReversed: false,
+    sortBy: '',
+  };
 
-type ReorderOptions = {
-  sortType: SortType,
-  isReversed: boolean,
-};
+  show = () => {
+    this.setState({ start: true });
+  };
 
-// Use this function in the render method to prepare goods
-export function getReorderedGoods(
-  goods: string[],
-  { sortType, isReversed }: ReorderOptions,
-) {
-  // To avoid the original array mutation
-  const visibleGoods = [...goods];
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  sortAlphabetically = () => {
+    this.setState({
+      sortBy: 'alpha',
+    });
+  };
 
-  return visibleGoods;
-}
+  sortByLength = () => {
+    this.setState({
+      sortBy: 'length',
+    });
+  };
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
+  reset = () => {
+    this.setState({
+      goods: goodsFromServer,
+      isReversed: false,
+      sortBy: '',
+    });
+  };
 
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
+  render() {
+    const {
+      start,
+      goods,
+      sortBy,
+      isReversed,
+    } = this.state;
+    const visibleGoods = [...goods];
 
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
+    if (sortBy) {
+      visibleGoods.sort((a, b) => {
+        switch (sortBy) {
+          case 'alpha':
+            return a.localeCompare(b);
+          case 'length':
+            return a.length - b.length;
+          default:
+            return 0;
+        }
+      });
+    }
 
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
+
+    return (
+      <div className={classNames('App', {
+        'App--before': !start,
+        'App--after': start,
+      })}
+      >
+        {!start
+          ? (
+            <button
+              type="button"
+              onClick={this.show}
+              className="button button--start"
+            >
+              Start
+            </button>
```

Comment:
For what this button? Maybe that's why the test doesn't pass, because it doesn't see the data at first?